LICENSE

```diff
@@ -199,3 +199,83 @@
+      maven-projects/spark/datasourcs-32/src/main/scala/org/apache/graphar/datasources/GarScanTable.scala
+      maven-projects/spark/datasourcs-32/src/main/scala/org/apache/graphar/datasources/GarWriteBuilder.scala
+      maven-projects/spark/datasourcs-32/src/main/scala/org/apache/graphar/datasources/csv/CSVWriterBuilder.scala
+      maven-projects/spark/datasourcs-32/src/main/scala/org/apache/graphar/datasources/orc/ORCOutputWriter.scala
+      maven-projects/spark/datasourcs-32/src/main/scala/org/apache/graphar/datasources/orc/ORCWriteBuilder.scala
+      maven-projects/spark/datasourcs-33/src/main/scala/org/apache/graphar/datasources/parquet/ParquetWriteBuilder.scala
+      maven-projects/spark/datasourcs-33/src/main/scala/org/apache/graphar/datasources/GarCommitProtocol.scala
+      maven-projects/spark/datasourcs-33/src/main/scala/org/apache/graphar/datasources/GarDataSource.scala
+      maven-projects/spark/datasourcs-33/src/main/scala/org/apache/graphar/datasources/GarScan.scala
+      maven-projects/spark/datasourcs-33/src/main/scala/org/apache/graphar/datasources/GarScanBuilder.scala
+      maven-projects/spark/datasourcs-33/src/main/scala/org/apache/graphar/datasources/GarScanTable.scala
+      maven-projects/spark/datasourcs-33/src/main/scala/org/apache/graphar/datasources/GarWriteBuilder.scala
+      maven-projects/spark/datasourcs-33/src/main/scala/org/apache/graphar/datasources/csv/CSVWriterBuilder.scala
+      maven-projects/spark/datasourcs-33/src/main/scala/org/apache/graphar/datasources/orc/ORCOutputWriter.scala
+      maven-projects/spark/datasourcs-33/src/main/scala/org/apache/graphar/datasources/orc/ORCWriteBuilder.scala
+      maven-projects/spark/datasourcs-33/src/main/scala/org/apache/graphar/datasources/parquet/ParquetWriteBuilder.scala
+    are modified from spark.
+
+* arrow 12.0.0 (https://github.com/apache/arrow)
+* fastFFI v0.1.2 (https://github.com/alibaba/fastFFI)
+    Files:
+      maven-projects/java/src/main/java/org/apache/graphar/stdcxx/StdString.java
+    are modified from fastFFI.
+
+* GraphScope v0.25.0 (https://github.com/alibaba/GraphScope)
+    Files:
+      maven-projects/java/src/main/java/org/apache/graphar/stdcxx/StdVector.java
+      maven-projects/java/src/main/java/org/apache/graphar/stdcxx/StdMap.java
+      maven-projects/java/src/main/java/org/apache/graphar/stdcxx/StdSharedPtr.java
+      maven-projects/java/src/main/java/org/apache/graphar/stdcxx/StdUnorderedMap.java
+      maven-projects/java/build.xml
+    are modified from GraphScope.
+
+================================================================
+MIT licenses
+================================================================
+The following components are provided under the MIT License. See project link for details.
+The text of each license is also included in licenses/LICENSE-[project].txt
+
+* result (https://github.com/bitwizeshift/result)
+    Files:
+      cpp/thirdparty/result.hpp is modified from result.
```

Comment:
It would be even better if we could include the version of the corresponding third-party code, licenses may change over time.

Reply:
Is that ok that `mini-yaml` and `simple-uri-parser` provide a commit hash since they did not release any version by now?

Reply:
absolutely ok:)